lib/BabelPluginExportToFunction.js

```diff
@@ -18,15 +18,28 @@ const isDefaultExport = R.allPass([
     R.pathEq([ 'object', 'name' ], 'exports'),
     R.pathEq([ 'property', 'name' ], 'default')
 ]);
+const hasMoreThanTwoParameters = R.pathSatisfies(R.gte(R.__, 2), [ 'params', 'length' ]);
 const isValidFunctionExpression = R.allPass([
     R.propEq('type', 'FunctionExpression'),
-    R.pipe(R.prop('params'), R.length, R.gte(R.__, 2))
+    hasMoreThanTwoParameters
+]);
+const isValidFunctionDeclaration = R.allPass([
+    R.propEq('type', 'FunctionDeclaration'),
+    hasMoreThanTwoParameters
 ]);
 const isIdentifier = R.propEq('type', 'Identifier');
 const getIdentifierNames = (functionDefinition) => R.pipe(
     R.map(buildArgumentIdentifier),
     R.append(CALLBACK_ID)
 )(R.range(0, functionDefinition.params.length - 2));
+const recursivelyResolveBinding = R.curry((bindings, identifier) => {
+    const resolved = bindings[identifier.name];
+    return R.ifElse(
+        R.pipe(R.propOr({}, 'init'), isIdentifier),
+        R.pipe(R.prop('init'), recursivelyResolveBinding(bindings)),
+        R.identity
+    )(resolved.path.node);
```

Comment:
FWIW: this creates a lot functions during each iteration of the recursion. But I wouldn’t expect a lot of iterations anyway.